samza-core/src/main/java/org/apache/samza/task/StreamOperatorTask.java

```diff
@@ -146,6 +147,13 @@ public final void processAsync(IncomingMessageEnvelope ime, MessageCollector col
               callback.complete();
             }
           });
+        } else {
+          // If InputOperator is not found in the operator graph for a given SystemStream, throw an exception else the
+          // job will timeout due to async task callback timeout (TaskCallbackTimeoutException)
+          final String errMessage = String.format("InputOperator not found in OperatorGraph for %s. Please check SystemStream "
```

Comment:
It would be helpful for debugging this in the case that the hash is different to log all available input operators to easily identify the mismatch using `operatorImplGraph#getAllInputOperators`

Reply:
Done.